metadata-builders/src/main/java/com/linkedin/metadata/builders/search/DatasetIndexBuilder.java

```diff
@@ -12,24 +12,59 @@
 import com.linkedin.metadata.search.DatasetDocument;
 import com.linkedin.metadata.snapshot.DatasetSnapshot;
 import com.linkedin.schema.SchemaMetadata;
-import lombok.extern.slf4j.Slf4j;
-
-import javax.annotation.Nonnull;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
 public class DatasetIndexBuilder extends BaseIndexBuilder<DatasetDocument> {
+  private static final Map<String, Character> PLATFORM_DELIMITER_MAP =
+      Collections.unmodifiableMap(new HashMap<String, Character>() {
+        {
+          put("adlsGen1", '/');
```

Comment:
Consider using an enum for the keys. Do we already have one in some model somewhere?

Reply:
platforms are plain strings today. They are not enums.

Reply:
We clearly have a well defined set of them. We should add an enum if one does not exist already.

Reply:
Or a set of actual constants, enum aside.

Point is, magic strings are bad.

Reply:
Platform entities are defined in a static json file https://github.com/linkedin/datahub/blob/master/gms/impl/src/main/resources/DataPlatformInfo.json 

On the contrary, we could make dataplatform name as enum and change the definition of Platform Urn. That can be a breaking change to call out.

Reply:
Why was this defined in JSON in the first place?

Reply:
It is the source of data for an immutable DAO.